### PR TITLE
Improve Test Coverage

### PR DESCRIPTION
**Pull request by mbabker**
The test coverage should be improved slightly with these additions.  Either methods or test cases have been added to try and cover cases in the code where existing tests missed a couple lines in a function.

### CODE REVIEW

File: tests/suite/joomla/application/JApplicationTest.php [hunk 1]
```diff
@@ -228,6 +237,38 @@ class JApplicationTest extends PHPUnit_Framework_TestCase
...
+	public function testGetHash()
+	{
+		$expResult = MD5("This is a test");
+		/**$mockApplication = $this->getMock('JApplication', array('getHash'), array('site'));
+		$mockApplication->expects($this->once())->method('getHash')->will(
+			$this->returnValue($expResult)
+		);**/
+		require_once(JPATH_PLATFORM . '/joomla/registry/registry.php');
+		$temp = JFactory::$application;
+		JFactory::$application = new JRegistry();
```
Comment (eddieajau) [inline tests/suite/joomla/application/JApplicationTest.php]: Wouldn't it be better to mock JFactory::getConfig here?  For example:

```
// Temporarily override the config cache in JFactory.
$temp = JFactory::$config;
JFactory::$config = new JObject(array('secret' => 'foo'));

$this->assertThat(
    JApplication::getHash('This is a test'),
    $this->equalTo(md5('foo'.'This is a test'),
    'Tests that the secret string is added to the hash.'
);

JFactory::$config = $temp;
```

Just a thought.

**Comment by eddieajau** [comment]
Would you mind adding @since 11.3 to any new tests that don't already have it?

**Comment by mbabker** [comment]
That test is actually a straight up copy of the one for the deprecated JUtility method.  I'm still learning how to build tests, so I'm sure it could be done better.

**Comment by eddieajau** [comment]
Ok, that's cool :)  I've got another pull in the queue to fix the unstable builds so I might wait for that to go through (I made it, so I can't merge it) before I pull this one.  Thanks though - every little bit helps.

**Comment by mbabker** [comment]
And I'll keep throwing commits on here in the mean time for every little bit I'm able to improve on.

**Pull request merged**